### DISCUSSION TITLE
Track PDF browser views and file downloads

### DISCUSSION THREAD

**Discussion by flaviamuntean** [plausible/analytics#877] [Anything Else]
## Prerequisites
- [x] I have searched open and closed issues to make sure that the bug has not yet been reported.

## Bug report
We've got a Gatsby site with several redirects which lead to a PDF. For example, if you go to `/privacy-policy` you are redirected to `/policies/Privacy_Policy.pdf`. The integration with Plausible was done using the [script](https://plausible.io/docs/plausible-script).

**Describe the bug**
It would appear that all of our usual pages, including redirects that lead to a regular page (i.e. `/team` redirects to `/careers`) are being tracked as expected and appear in the Plausible dashboard. However, redirects that lead to a .pdf page do not appear in the dashboard. This is something that used to work when using Google Analytics.

**Expected behavior**
I would expect to see `/policies/Privacy_Policy.pdf` listed in the Plausible dashboard.

**Update:** You can now track file downloads! All the most common file extensions are supported but you can also specify custom file types to track. Here's how to enable file downloads tracking https://plausible.io/docs/file-downloads-tracking

**Comment by metmarkosaric** [comment]
thanks @flaviamuntean! moving this over to the forum as this is more of a feature request rather than an issue. as long as you don't have our tracking script embedded on a specific page, we cannot track it which means PDFs are not tracked. 

we could for instance track file downloads in the future which should cover this use case. at the moment, you could use our custom events to track clicks that lead to the PDF file

**Comment by metmarkosaric** [comment]
After never hearing about this, we've now had this feature request to track PDF browser views and file downloads three times in a week

**Reply by ukutaht** [reply]
I am struggling to find information about how GA tracks file downloads automatically. Every resource I've found is a tutorial on how to set up event triggers etc.

An example would be good that I can debug. Is there a publicly available site/page where file downloads are tracked automatically with no extra config with GA? I'd love to see one.

**Reply by metmarkosaric** [reply]
File downloads and external link clicks are the new options in GA4 but for the old GA it doesn't work out of the box. There are many extensions or guides that make this simple to enable so anyone that wants it can do it. 

The expectation is not necessary that this works out of the box in Plausible but could be a nice "easy guide" addition similar to how we have 404 page tracking and external link click tracking in our docs for those who want to track file downloads / PDF. There's no easy way to discover those instructions for Plausible for those who don't code while for GA it's pretty simple to find many guides.

**Reply by ukutaht** [reply]
Hmm from this

> This is something that used to work when using Google Analytics.

I got the impression that this worked out out of the box even with old GA. I probably misunderstood.

I can totally see a `plausible.downloads.js` extension to get automatic tracking for file downloads. 

I checked the GA guides and it seems that the more techincally correct term would be tracking 'download links' instead of 'file downloads'.

If the link to the download is shared somewhere outside your website, I'm not aware of a way to track that download. So maybe it would be called `plausible.download-links.js` instead to make sure the customer has the correct idea of what is actually being tracked.

**Reply by metmarkosaric** [reply]
Could also be people set it up and then forget that they manually added a piece of code to make it work.

Ha interesting! So a visit from say HN directly to a PDF file cannot be tracked. It can only be tracked if the visitor actually is on your site first and clicks on a link to the PDF on your site.

**Reply by ukutaht** [reply]
> Ha interesting! So a visit from say HN directly to a PDF file cannot be tracked. It can only be tracked if the visitor actually is on your site first and clicks on a link to the PDF on your site.

Yeah I cannot see a way you could add a JS tracker or a pixel on a file download itself. Just doesn't make sense to me but I could be wrong.

**Reply by ksdhans** [reply]
I'd also like an easy way to track file download link clicks. A plausible.download-links.js script would be great. 

I wish that the CDN would provide actual download data, but Cloudflare doesn't (at least, not the free/low tiers).

**Comment by metmarkosaric** [comment]
You can now track file downloads! All the most common file extensions are supported but you can also specify custom file types to track. Here's how to enable file downloads tracking https://plausible.io/docs/file-downloads-tracking

**Reply by ksdhans** [reply]
Thanks.

Is there a way to add to the default file-types instead of replacing the list? Or, could we specify a wildcard so all file extensions are tracked?

**Reply by metmarkosaric** [reply]
you're welcome! you could add the list of types that we track by default to your custom list. @RobertJoonas might know if there's some nicer looking way of doing it

**Reply by RobertJoonas** [reply]
Currently, the only way to achieve this is to copy all the default file types into the `file-types` attribute and then add your custom ones as well. I only made it work with one attribute to keep it simple, but I guess we could add an additional `add-file-types` (not sure if the best name for it) attribute that would also keep the defaults.
